SharedSource/Core/Utilities/Assertion.cs

```diff
@@ -16,8 +16,9 @@
 // 
 using System;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using JetBrains.Annotations;
-
+#nullable enable
 // ReSharper disable once CheckNamespace
 namespace Remotion.Utilities

```

Comment:
`IsTrue` and `IsFalse` need `DoesNotReturnIfAttribute` for the assertion condition.

e.g.:
```csharp
[AssertionMethod]
public static void IsTrue ([DoesNotReturnIf (false), AssertionCondition (AssertionConditionType.IS_TRUE)] bool assertion, string message)
{
  IsTrue (assertion, message, s_emptyArguments);
}
```